aptos-move/framework/src/natives/aggregator_natives/helpers_v2.rs

```diff
@@ -1,11 +1,55 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
-use crate::natives::aggregator_natives::helpers::get_aggregator_field;
-use aptos_aggregator::aggregator_extension::extension_error;
-use move_binary_format::errors::PartialVMResult;
+
+use crate::natives::aggregator_natives::helpers_v1::get_aggregator_field;
+use aptos_aggregator::aggregator_extension::{extension_error, AggregatorID};
+use move_binary_format::errors::{PartialVMError, PartialVMResult};
+use move_core_types::vm_status::StatusCode;
 use move_vm_types::values::{Struct, StructRef, Value};
 
+/// Indices of `value` and `limit` fields in the `Aggregator` Move
+/// struct.
 const VALUE_FIELD_INDEX: usize = 0;
+const LIMIT_FIELD_INDEX: usize = 1;
+
+pub const EINVALID_AGGREGATOR_IDENTIFIER: u64 = 0x02_0008;
+
+/// Returns ID and a limit of aggrgegator based on a reference to `Aggregator` Move struct.
+pub(crate) fn aggregator_value_as_u128(
+    aggregator: &StructRef,
+) -> PartialVMResult<(AggregatorID, u128)> {
+    let (value, limit) = get_aggregator_fields_u128(aggregator)?;
+    if value > u64::MAX as u128 {
+        return Err(PartialVMError::new(StatusCode::ABORTED)
+            .with_message("Aggregator identifier is too small".to_string())
+            .with_sub_status(EINVALID_AGGREGATOR_IDENTIFIER));
```

Comment:
You can use `SafeNativeError::InvariantViolation(...)` here as well? Or simply invariant violation PartialVMResult.

Reply:
Done.